tests/store/_unity_catalog/model_registry/test_unity_catalog_rest_store.py

```diff
@@ -0,0 +1,189 @@
+from itertools import combinations
+
+import json
+import pytest
+from unittest import mock
+
+from mlflow.entities.model_registry import RegisteredModelTag
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_uc_registry_messages_pb2 import (
+    CreateRegisteredModelRequest,
+    UpdateRegisteredModelRequest,
+    DeleteRegisteredModelRequest,
+    GetRegisteredModelRequest,
+    SearchRegisteredModelsRequest,
+)
+from mlflow.store._unity_catalog.registry.rest_store import UcModelRegistryStore
+from mlflow.utils.proto_json_utils import message_to_json
+from mlflow.utils.rest_utils import MlflowHostCreds
+from tests.helper_functions import mock_http_200
+
+
+def host_creds():
+    return MlflowHostCreds("https://hello")
+
+
+@pytest.fixture
+def store():
+    return UcModelRegistryStore(host_creds)
```

Comment:
Strange. Why can we instantiate `UcModelRegistryStore`? It doesn't implement all the abstract methods.

Reply:
I think `BaseRestStore` shoud look like:

```python
class BaseRestStore(AbstractStore):
    def __init__(self, get_host_creds):
        self.get_host_creds = get_host_creds

    def _call_endpoint(self, api, json_body, call_all_endpoints=False):
        response_proto = self._get_response_from_method(api)
        if call_all_endpoints:
            endpoints = self._get_all_endpoints_from_method(api)
            return call_endpoints(self.get_host_creds(), endpoints, json_body, response_proto)
        else:
            endpoint, method = self._get_endpoint_from_method(api)
            return call_endpoint(self.get_host_creds(), endpoint, method, json_body, response_proto)

    @abstractmethod
    def _get_response_from_method(self, method):
        pass

    @abstractmethod
    def _get_all_endpoints_from_method(self, method):
        pass

    @abstractmethod
    def _get_endpoint_from_method(self, method):
        pass

    # CRUD API for RegisteredModel objects

    @abstractmethod
    def create_registered_model(self, name, tags=None, description=None):
        pass

    @abstractmethod
    def update_registered_model(self, name, description):
        pass
    
    # Other methods
    ...
```

Reply:
With `BaseStore` above, we'll get `TypeError: Can't instantiate abstract class UcModelRegistryStore with abstract methods create_model_version, delete_model_version, delete_model_version_tag, get_model_version, get_model_version_download_uri, search_model_versions, set_model_version_tag, transition_model_version_stage, update_model_version`. To suppress this error, we can use a dummy implementation:

```python
    def delete_model_version_tag(self, name, version, key):
        raise NotImplementedError("TODO")
```

Reply:
@harupy thanks for the find! Your suggestion seems like the intended behavior + I like that it actually enforces that abstract methods are implemented. The only issue I think is that the change will break plugins that have been developed against the existing AbstractStore interface and rely on the existing behavior of being able to instantiate AbstractStore subclasses without implementing all the abstract methods.

Arguably these plugins are invalid & should be fixed, although we also have added new APIs since the original plugin interface release in https://github.com/mlflow/mlflow/pull/1943, e.g. APIs for tags on registered models and versions (so we might need to add a default, throwing implementation of the tagging APIs in AbstractStore if we want to enforce that subclasses implement all abstract methods). But maybe it’s something we can consider for a follow-up PR? I would still hesitate to make a breaking behavior change to downstream plugins before MLflow 3.0